{% extends "layout/layout.html" %}


{% block content %}
{{ heading("Parts and Materials Registry") }}
<style>
  table, th, td {
    border: 1.5px solid white;
    background-color: #285a06;
    color: white;
  }
  th, td {
    padding: 1em;
  }
  thead {
    background-color: #285a06;
  }
  tbody {
    background-color: #dfffc9;
    color: black;
  }

  ul {
    list-style-type: disc;
    list-style-position: outside;
    padding-left: 2em;
  }
  caption { 
    caption-side: bottom;
    text-align: left;
    margin-top: 10px;
    color: black;
    font-style: italic
  }
</style>
<div class="custom">
  <h4>Table 1: Bacterial Strains Utilised</h4>
  <div class="lg:overflow-visible overflow-scroll">
    <table class = "font-jost" style = "width:100%">
        <caption>
          This table lists the bacterial strains used in the study, along with their descriptions and sources.
        </caption>
        <thead>
            <tr>
                <th>Strain Name</th>
                <th>Description</th>
                <th>Source</th>
              </tr>
        </thead>
        <tbody>
            <tr>
                <td>E. coli TB18</td>
                <td><ul>
                    <li>E. coli T-B18 is a genetically engineered, adaptively evolved strain used in recent methylotrophy studies</li>
                    <li>Has frmA gene knocked out</li>
                    <li>Has pETM6_Ptrc_BsMdh_BmHps_BmPhi plasmid introduced</li>
                    </ul></td>
                <td><u><a href="https://www.addgene.org/159446/">https://www.addgene.org/159446/</a></u></td>
              </tr>
              <tr>
                <td>E. coli DH5α</td>
                <td>
                    <ul>
                        <li>
                            E. coli DH5α is an expression strain for propagating plasmids.
                        </li>
                    </ul>
                <td><u><a href="https://www.neb.com/en/products/c2987-neb-5-alpha-competent-e-coli-high-efficiency#Product%20Information">
                    https://www.neb.com/en/products/c2987-neb-5-alpha-competent-e-coli-high-efficiency#Product%20Information</a></u></td>
              </tr>
        </tbody>
      </table>
  </div>
  <p><br /></p>
  <h4>Table 2: Vectors used in cloning and CRISPR-Cas9 Knockout</h4>

  <div class="lg:overflow-visible overflow-scroll">
    <table class = "font-jost" style = "width:100%" borders = "">
      <caption>
        The vectors utilized in the cloning procedures and CRISPR-Cas9 genome editing techniques, including their specific roles and sources.  
      </caption>
      <thead>
          <tr>
              <th>Vector Name</th>
              <th>Description</th>
              <th>Source</th>
            </tr>
      </thead>
      <tbody>
          <tr>
              <td>pETM6</td>
              <td><ul>
                  <li>pETM6_Ptrc_BsMdh_BmHps_BmPhi (pETM6) expresses genes (Mdh, Hps, and Phi) conducive for methylotrophy.</li>
                  </ul></td>
              <td><u><a href=https://www.addgene.org/129104>https://www.addgene.org/129104</a></u></td>
            </tr>
            <tr>
              <td>pTargetF</td>
              <td><ul>
                  <li>Contains the sgRNA sequence complimentary to the desired 5’-NGG-3’ site for knockout.
                  </li>
                  </ul></td>
              <td><u><a href=https://www.addgene.org/62226>https://www.addgene.org/62226</a></u></td>
            </tr>
            <tr>
              <td>pCas9</td>
              <td><ul>
                  <li>Contains genes encoding the Cas9 complex necessary for performing CRISPR-Cas9 knockouts
                  </li>
                  </ul></td>
              <td><u><a href=https://www.addgene.org/42876>https://www.addgene.org/42876</a></u></td>
            </tr>
            <tr>
              <td>Circularized donor DNA</td>
              <td><ul>
                  <li>Composed of the upstream and downstream homology arms to T-B18’s tpiA gene.
                  </li>
                  <li>Replaces tpiA via homology-directed repair.</li>
                  </ul></td>
              <td>[N/A]</td>
            </tr>
      </tbody>
    </table>
  </div>
  <p><br /></p>
  <h4>Table 3: Oligos used in cloning and CRISPR-Cas9</h4>
  <div class="lg:overflow-visible overflow-scroll">
  <table class = "font-jost" style = "width:100%" borders = "">
      <caption>
        A list of oligonucleotides used in cloning and CRISPR-Cas9 processes, featuring their sequences and intended purposes in the experiments
      </caption>
      <thead>
          <tr>
              <th>Oligo name</th>
              <th>5' - 3' sequence</th>
              <th>Purpose</th>
            </tr>
      </thead>
      <tbody>
          <tr>
              <td>pCas9_Cas9_F</td>
              <td>CTCAATAGGCTTAGATATCGGCAC</td>
              <td><ul>
                  <li>Forward primer used to sequence the Cas9 gene in pCas9.</li>
                  </ul></td>
            </tr>
            <tr>
              <td>pCas9_Cas9_R</td>
              <td>CTTCAGTCACCTCCTAGCTGAC</td>
              <td><ul>
                  <li>Reverse primer used to sequence the Cas9 gene in pCas9.
                  </li>
                  </ul></td>
            </tr>
            <tr>
              <td>pCas9_CmR_F</td>
              <td>GGAGCTAAGGAAGCTAAAATGG</td>
              <td><ul>
                  <li>Used to sequence the chloramphenicol resistance cassette in pCas9.
                  </li>
                  </ul></td>
              
            </tr>
            <tr>
              <td>sgRNA_F</td>
              <td>CAAGTGATCATTCAGTACGTTTTTTTGAATTCTCTAGAGTCGACCTGCAGAAGCT</td>
              <td><ul>
                  <li>Forward primer used for inverse PCR of pTargetF to insert desired sgRNA sequence
                  </li>
                  </ul></td>
            </tr>
            <tr>
              <td>sgRNA_R</td>
              <td>GTACTGAATGATCACTTGGCACCGACTCGGTGCCACTTTTTC</td>
              <td><ul>
                  <li>Reverse primer used for inverse PCR of pTargetF to insert desired sgRNA sequence.
                  </li>
                  </ul></td>
            </tr>
            <tr>
              <td>ΔtpiA_cPCR_F</td>
              <td>CTGTATAAAAACGTCGAAGTTCTGGATTCTGGC</td>
              <td><ul>
                  <li>Forward colony PCR primer used for verification of tpiA knockout.
                  </li>
                  </ul></td>
            </tr>
            <tr>
              <td>ΔtpiA_cPCR_R</td>
              <td>GTGCCAACAAACGCACCAAGATCGATAA</td>
              <td><ul>
                  <li>Reverse colony PCR primer used for verification of tpiA knockout.
                  </li>
                  </ul></td>
            </tr>
            <tr>
              <td>sgRNA_F frdA</td>
              <td>CGATGAACTCTGGGTTCAGGCC</td>
              <td><ul>
                  <li>Forward primer to insert 20 bp of sgRNA of frdA into pTargetF.
                  </li>
                  </ul></td>
            </tr>
            <tr>
              <td>sgRNA_R frdA</td>
              <td>CGTGTCTCAAACGGGACCAAATGAATATCGG</td>
              <td><ul>
                  <li>Reverse primer to insert 20 bp of sgRNA of frdA into pTargetF.
                  </li>
                  </ul></td>
            </tr>
      </tbody>
  </table>
  </div>
  <div>
    <p>
      The two gRNA primers were obtained by introducing 20 bp of sgRNA within the tpiA genome that guides the Cas9 protein to cut and create strand breaks.The 20bp of sgRNA was generated using CHOPCHOP and analyzed based on various factors such as GC content, self-complementarity and efficiency. 
      Collectively, they will amplify the backbone of pTargetF along with targeted tpiA sgRNA sequence to generalize linear PCR fragments, products will be transformed into E.coli later and self-circulaized in T-B18. 
    </p>
    <p>
      The tpiA Donor DNA is 1000 bp of synthesized double strand DNA that has 500 bp homologous to both upstream and downstream of tpiA gene region in T-B18 genome.
      Served as repair template for homologous recombination after Cas9 endonucleases cut at frdA region to knockout tpiA gene.
    </p>
    <p>
      Both tpiA colony PCR primers were acquired by designing primers that are complementary to 500bp both upstream and downstream of the tpiA gene in E.coli genome.
      We will use these two parts to test the knockout results of the tpiA gene; if the products are longer than 1000bp, the tpiA is not knocked out successfully. If the amplified products are 1000bp, the tpiA gene is knockout successfully.
    </p>
    <p>
      Both frdA gRNA primers were obtained by introducing 20 bp of sgRNA within the tpiA genome that guides the Cas9 protein to cut and create strand breaks.The 20bp of sgRNA was generated using CHOPCHOP and analyzed based on various factors such as GC content, self-complementarity and efficiency. 
      This part will amplify the backbone of pTargetF along with targeted frdA sgRNA sequence to generalize linear PCR fragments, products will be transformed into E.coli later and self-circulaized in T-B18.
    </p>
  </div>

  <h4>Table 4: Apparatus used in Rational Design.</h4>
  <div class="lg:overflow-visible overflow-scroll">
    <table class = "font-jost" style = "width:100%" borders = "">
      <caption>
        Catalog of key apparatus employed in rational design experiments, outlining their specific uses in the analysis.
      </caption>
      <thead>
          <tr>
              <th>Name</th>
              <th>Purpose</th>
            </tr>
      </thead>
      <tbody>
          <tr>
              <td>ICS5000 (HPLC machine)</td>
              <td><ul>
                  <li>Used to run HPLC on bacterial samples grown overnight in methanol-containing minimal media to measure consumption.</li>
                  </ul></td>
            </tr>
            <tr>
              <td>GC Machine</td>
              <td>
                  <ul>
                      <li>
                          Used as an alternative quantification apparatus to the ICS5000.
                      </li>
                      <li>
                          Used to run gas chromatography on bacterial samples grown overnight in methanol-containing minimal media to measure consumption.
                      </li>
                  </ul>
            </tr>
      </tbody>
    </table>
  </div>
</div>

{% endblock %}