{% extends "layout/layout.html" %}

{% block content %}
<div>
  {{subheading("Introduction")}}
  <div>
    <p>
      The goal of our project was to engineer a bacterial strain which preferentially assimilates methanol over other carbon sources. Overall, preliminary results showed that our T-B18 E. coli strain metabolizes glucose over methanol, suggesting that additional genetic modifications are required. Although our aim was not met, our results do highlight several pitfalls, potential paths to explore further, and factors to consider for effectively developing a methanol-consuming bacteria for use in landfill applications.
    </p>
  </div>
  {{img(cdn("pictures/wl-results-fin.png"), alt = "Fig. 1: Overall Results Workflow")}}
</div>
<div>
  {{ subheading("1. Plasmid curing of pETM6 from T-B18")}}
  <div>
      {{ subsubheading("Step 1: Prepare T-B18 Glycerol Stocks") }}
      <p>
          T-B18 was ordered from Addgene and it arrives as a bacterial stab. We need to streak the bacterial stab onto plates. T-B18 has KAN  25 μg/mL resistance so we will make LB agar plates with CAM. Bacterial stab is streaked so that individual colonies can be picked. Once we have individual colonies on a plate, we will pick a colony to inoculate and make a glycerol stock with. This step was completed for us by Ethan Agena. 3 T-B18 glycerol stocks were made and stored at -80°C.
      </p>
  </div>
  <div>
      {{ subsubheading("Step 2: Test T-B18 for Rifampicin (RIF) Sensitivity") }}
      <p>
        RIF can be used to cure plasmids by adding stress to the cells while in the media.[5] This curing method only works if the organism/strain is sensitive to RIF so before we can start with RIF curing, we must test its sensitivity to RIF. T-B18 was first cultured LB + KAN + AMP, then plated on 2 different plates. One plate has RIF at a concentration of 50 μg/mL while the other plate has no RIF.   
      </p> 
      <div class="side-by-side">
          {{img(cdn("pictures/wl-curing2.png"), alt = "Fig. 2: Curing Plates")}}
          <p>
              We observed colonies growing on an LB only plate (left) while no colonies were observed on the plate containing RIF, confirming that T-B18 is sensitive to RIF. Thus, we can proceed with the RIF plasmid curing method.
          </p>
      </div>
  </div>
  <div>
    {{ subsubheading("Step 3: RIF Plasmid Curing")}}
    <p>
      The plasmid curing protocols were adapted from Obaseiki-Ebor, 1984. The curing process relies on using RIF, which inhibits RNA polymerase from synthesizing RNA. This inhibits the replication of the plasmid, allowing the plasmid to be cured out after a few generations. 
    </p>
    <div class="side-by-side">
      {{img(cdn("pictures/wl-curing3.png"), alt = "Fig. 3: Replica Plates")}}

      <p>
        Out of the 4 treatments, only the 4hr 10 μg/mL RIF plate had distinct colonies so it was used to replica plate onto Kanamycin (KAN) + Ampicillin (AMP) plates. pETM6 carries an AMP resistance cassette so if the plasmid is cured, it will lose AMP resistance.
      </p>
    </div>
    <P>
      We found 5 colonies that lacked AMP resistance so we picked the colonies from the master plate to proceed with inoculation into LB + KAN + AMP to confirm loss of antibiotic resistance.
    </P>
    <div class="side-by-side">
      {{img(cdn("pictures/wl-master.png"), width="50%", alt="Fig. 4: Master Plate")}}
    </div>
    <div class="side-by-side">
      {{img(cdn("pictures/wl-replica1.png"), alt="Fig. 5: Replica Plate #1")}}
      {{img(cdn("pictures/wl-replica11.png"), alt="Fig. 6: Replica Plate #1 compared to Master")}}
    </div>
    <div class="side-by-side">
      {{img(cdn("pictures/wl-replica2.png"), alt="Fig. 7: Replica Plate #2")}}
      {{img(cdn("pictures/wl-replica22.png"), alt="Fig. 8: Replica Plate #2 compared to Master")}}
    </div>
    <p>
      All 5 colonies were confirmed to have lost resistance.
    </p>
    <div class="side-by-side">
      {{img(cdn("pictures/wl-cp-curing.png"),alt = "Fig. 9: Colony PCR Results", width="120%")}}
    </div>
    <p>
      A colony PCR was performed to confirm the absence of the pETM6 plasmid. All 5 colonies were cured of pETM6.
    </p>
  </div>
</div>
<div>
  {{ subheading("2. Replacement of B. m Mdh with C. n Mdh in pETM6 ") }}
  <p>
    The experimental workflow for this experiment involved the plasmid miniprep of B.m Mdh, the restriction digestion using enzymes, the ligation and transformation. 
    The plasmid miniprep was executed effectively, however during the extraction phase, the team was unable to verify the results due a concentration that was repeatedly too low. Therefore, given the timeframe of the experiment, the team conlucded that the results for the Mdh replacment were inconclusive and the experiment was unsuccessful. 
  </p>
</div>
<div>
  {{ subheading("3. CRISPR-Cas9 Knockout") }}
  <div>
    <p>
      In this investigation, we employed a genetic modification approach based on the CRISPR-Cas9 system to target the tpiA gene in E. coli strain T-B18.  
    </p>
    <p>
      The methodology involved the incorporation of a single guide RNA (sgRNA) sequence into the tpiA gene locus using inverse PCR, enabling precise Cas9-mediated double-strand break (DSB) induction. We transformed the pCas9 plasmid into cured T-B18 from the previous step, and after the pCas9-gRNA complex created a DSB, homology dependent repair was accomplished via the introduction of donor DNA that lacked the tpiA gene sequence, effectively rendering the gene nonfunctional.<br>
      To verify the knockout's success, we designed specific primers flanking 500 base pairs up and downstream of the tpiA gene. Colony PCR was conducted with these primers, resulting in PCR products with an expected size of approximately 1 kilobase (kb). Subsequent gel electrophoresis analysis confirmed the presence of bands matching the anticipated 1 kb length,  providing evidence that the tpiA gene had been knocked out. According to the gel image, lane 1 was loaded with a size marker–the 4th band from bottom represent 1kb, lane 2-6 were loaded with colony PCR results, lane 4 and land 5 display relatively pure 1 kb long PCR product. The corresponding colonies were selected to generate a pure tpiA knock-out stock for further experiments.     
    </p>
    {{img(cdn("pictures/screenshot-2023-10-11-at-9-45-40-pm.png"),alt="Fig. 10: Gel Electrophoresis results verifying the tpiA knockout.")}}
  </div>
</div>

<div>
  {{ subheading("4. Quantification using HPLC") }}
  <div>
    <p>
      To determine whether T-B18 (our methylotrophic strain) exhibited any change in methylotrophic tendencies after genetic modification, we performed HPLC on cultures grown for exactly 24 hours in methanol (MeOH) containing media. Four samples were prepared and run through the ICS5000 HPLC machine:
    </p>
    <p>
        <ul style="list-style-type:disc">
            <li>Water: indicates the presence of noise and impurities in column.</li>
            <li>M9 minimal media with 0.1M MeOH: used as a reference for the MeOH peak shape (negative control)
            </li>
            <li>T-B18 + pETM6: our baseline strain which already consumes methanol (positive control)
            </li>
            <li>T-B18_ΔtpiA: our genetically modified T-B18 mutant that should consume MeOH with increased efficiency compared to the baseline (treatment)
            </li>
        </ul> 
    </p>
    <p>
        To ascertain whether the T-B18 strains consumed MeOH, we compared the area under the refractive index (RI) chromatograms generated by passing our samples through the absorption column (shown below). This area is directly correlated to concentration via a standard MeOH curve. A peak of reduced width and amplitude compared to the negative control (M9 + MeOH) represents a smaller concentration of MeOH in the sample, which indicates that MeOH was consumed over the 24 hour growth period.
    </p>
    {{img(cdn("pictures/wl-hplc-results.png"),alt="Fig. 11: Chromatograms of HPLC samples.")}}
    <p>
      The size of the rightmost peak for the upper right, upper left, and lower left graphs represents the amount of MeOH present in the sample. The middle peaks represent the amount of glucose present in the samples. The clean chromatogram for the water sample indicates a clean column with little noise and impurities.
    </p>
    <p>
      There is no statistically significant difference in the total peak size from the above chromatograms to indicate that any of the T-B18 strains consumed MeOH. This is unexpected, as T-B18 + pETM6 is known to be methylotrophic. However, each strain consumed glucose (indicated by the chromatograms’ smaller middle peak sizes). This trend suggests that the T-B18 strains might have preferentially metabolized glucose over MeOH.
    </p>
    <p>
      In future experiments, we plan to grow cells on M9 media with reduced glucose content and observe whether their MeOH consumption changes.
    </p>
  </div>
</div>

{% endblock %}
